### ISSUE TITLE
Release 0.11.0

### ISSUE THREAD

**Issue by hacdias** [ipfs-shipyard/ipfs-desktop#1415]
# Release Notes

- Web UI: [`v2.8.0`](https://github.com/ipfs-shipyard/ipfs-webui/releases/v2.8.0)
- IPFS: [`go-ipfs v0.5.0-rc4`](https://github.com/ipfs/go-ipfs/releases/tag/v0.5.0-rc4)

### 🚨 Breaking Changes

- fix: remove old auto launch method (#1416)
- fix: ipfs daemon flags (#1437)

### ✨  Features

- feat: support for go-ipfs 0.5 (#1392) 
- feat: run garbage collector (#1407)
- feat: update ipfsd-ctl to 4.x (#1411)
- feat: improved update experience (#1414)
- feat(ux): improve download cid (#1419)
- feat(ux): improve gc (#1420)
- feat(ux): better error messages (#1421)
- feat(ux): separate open from other actions (#1424)
- feat(ux): move preferences to menubar (#1425) 
- feat: allow custom ipfs binary (#1427)
- feat(ux): open web ui at login (#1429)
- feat: add "Help Translation" under "About" (#1447)
- feat: go-ipfs 0.5 (https://github.com/ipfs-shipyard/ipfs-desktop/pull/1463)

### 🛠 Fixes and Maintenance

- fix: drag&drop files and folders (https://github.com/ipfs-shipyard/ipfs-desktop/issues/1287#issuecomment-620604299)
- fix: start on login (#1334)
- fix: notify when new update is available (#1384)
- chore: update to electron 8.x (#1404)
- fix: write path when it does not exist (#1405)
- refactor: remove babel (#1406)
- fix: circular dependency between dialogs and daemon (#1408)
- fix: use portfinder instead of get-port (#1410)
- fix: screenshot taking (#1418)
- fix: disable updates in development (#1423)
- docs: clarify protocol usage (#1426)
- fix: translate 'Select Directory' (#1428)
- refactor: move utilities to utils directory (#1434)
- fix: repo fsck is deprecated (#1438)
- refactor: use store migrations and defaults (#1443)
- fix(ux): show dock when prompt is visible (#1450)
- fix: keep track of last opened page (#1452)

# Before Release

- [x] Make sure i18n is updated (`tx pull -a`)
- [x] Make sure we're using the latest released version of Web UI
- [x] Make sure we're using the latest released version of `go-ipfs`
- [x] Update Release Notes accordingly

# Release

- [x] Release Candidate
	- [x] 💻 Bump version on `package.json`
	- [x] 💻 `git tag $version`
	- [x] 💻 `git push && git push --tags`
	- [x] ⏰ Wait...
	- [x] 🔗 Link release notes to this issue
- [ ] Final
	- [x] 📝 Create draft and copy the release notes above
	- [x] 💻 Bump version on `package.json`
	- [x] 💻 `git tag $version`
	- [x] 💻 `git push && git push --tags`
	- [x] ⏰ Wait...
	- [x] 📃 Copy the release notes above to the GitHub release
    - [ ] :police_officer:  Notarize `.dmg` at Apple (context: [#1365](https://github.com/ipfs-shipyard/ipfs-desktop/issues/1211))
          1. Download `.dmg` from `https://github.com/ipfs-shipyard/ipfs-desktop/releases/vA.B.C`
          2. Ensure `APPLEID` and `APPLEIDPASS` are set either as environment variables or entries in `.env` file. Those need to belong to the same org as cert used for signing.
         3. Run `node pkgs/macos/notarize-cli.js ./IPFS-Desktop-A.B.C.dmg`
	- [ ] 🚢 Ship!
	- [ ] 📑 Update README.md versions
	- [ ] 📦 Update Homebrew
	- [ ] 📦 Update Chocolatey

# Postponed

_If any..._

**Comment by lidel** [comment]
Note:  [update to electron 8](https://github.com/ipfs-shipyard/ipfs-desktop/commit/5afb16c2c89226f58ad21bab8e1bf6d408a9d3c8#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R62) broke tray icon on linux, making it impossible to start/stop the node, nor open webui  – see https://github.com/ipfs-shipyard/ipfs-desktop/issues/1153#issuecomment-596780147 

Releasing this would introduce regression for existing users.
 We need to either revert that update, or figure out a way of solving the Linux problem.

**Comment by hacdias** [comment]
Hmm. You’re right. Honestly I doubt Electron/Chromium folks are going to fix this soon. It’s been years talking about Linux and its Tray icons and nothing has been done. 

One one hand, I don’t want to prevent Linux users to have access to IPFS desktop. On the other hand, I don’t want to keep using an old version of Electron ad infinitum when there are no assurances that this will be fixed. 

I’ll revert this for now (will do tomorrow) but we need to find a solution.

**Comment by lidel** [comment]
I agree, but I don't believe we should revert this: getting the latest Electron fixes are probably more important than keeping legacy tray support on Linux.

We just need to have UX workaround in place for Linux before 0.11 ships (show webui when `ipfs-desktop` is called while it is already running, or could be a custom popup, could be extensive "Tray on Linux" section in README etc – need to think about)

**Comment by hacdias** [comment]
Okay, let's find out a way!

**Comment by momack2** [comment]
Hey Folks! Flagging this for the go-ipfs 0.5 release train. go-ipfs RC-4 (https://github.com/ipfs/go-ipfs/issues/7109#issuecomment-619351992) has the final changes prior to release next week. We'd like the network to update ASAP to 0.5 so that everyone can experience the performance improvements. If this desktop release is currently blocked on the Linux issues mentioned above, can we cut an intermediary release which will help upgrade desktop users to 0.5.0 next week?

**Comment by momack2** [comment]
Should there be anything in these release notes about MacOS notarization too? (https://github.com/ipfs-shipyard/ipfs-desktop/pull/1365#issuecomment-596727138)

**Comment by hacdias** [comment]
@momack2 #1429 should unblock this. Just waiting for review! 😄

**Comment by lidel** [comment]
Notarization is still a manual step that needs to happen before publishing release draft on github. I added it to the checklist before "Ship" step, to ensure we don't forget.

**Comment by lidel** [comment]
Ok, I believe we reviewed and merged key PRs (Linux is unblocked by  #1429).

Things that need to happen before release:
- [] @rafaelramalho19 is  smoke-testing testing `master` with go-ipfs 0.5.0-rc4 (on Mac&Windows)
- [ ] switch to a new release version of ipfs-webui (https://github.com/ipfs-shipyard/ipfs-desktop/issues/1451#issuecomment-619940184)
  - we need a new release, as there were fixes that will impact desktop
    - latest `master` of webui includes  drag&drop fix by @rafaelramalho19 and may solve issues described in   https://github.com/ipfs-shipyard/ipfs-desktop/issues/1287#issuecomment-580368657
    - ideally, we would  wait for https://github.com/ipfs-shipyard/ipfs-webui/pull/1444, but may ship something befor that PR is properly reviewed and  merged 
- [ ] switch to go-ipfs 0.5.0 (when released - tracked in https://github.com/ipfs/go-ipfs/issues/7109)
- [ ] release ipfs-desktop & notarize on macOS (with `notarize-cli.js`, as noted in https://github.com/ipfs-shipyard/ipfs-desktop/pull/1365#issuecomment-596727138)

**Comment by lidel** [comment]
We switched to ipfs-webui 2.8.0 and drag&drop issues should be gone.

 @rafaelramalho19 is taking a look at the remaining problem with adding folders via button (drag&drop works: https://github.com/ipfs-shipyard/ipfs-desktop/issues/1287#issuecomment-620604299). 

Would be nice to include a fix for that in this release, but if the fix takes longer we can always ship it later as a patch release (it was already broken in ipfs-desktop v0.10.4 too, so no regression for existign users) – priority should be switching desktop users to go-ipfs 0.5.

**Comment by bertrandfalguiere** [comment]
I'm currently correcting translations for IPFS Desktop and the Web UI, if you can afford to wait 1h.

**Comment by bertrandfalguiere** [comment]
Done. You may want to reimport translations for Web UI, Desktop, Share files, Companion and IPLD explorer.

**Comment by lidel** [comment]
Thank you @bertrandfalguiere!

@jessicaschilling @rafaelramalho19 
I checked in with go-ipfs teams and we are prioritizing shipping go-ipfs 0.5 to users, so we will cut 0.11.0 today with go-ipfs 0.5 and then ship a patch release of Desktop (0.11.1) with fix for https://github.com/ipfs-shipyard/ipfs-desktop/issues/1287#issuecomment-620604299 and any missed translation updates.

**Comment by lidel** [comment]
(tagged and built v0.11.0, waiting for macOS notarization before publishing release)

**Comment by lidel** [comment]
Applied two hotfixes, recreating release:
- fix: ipfs on PATH on windows with spaces in binary path #1465
- fix: go-ipfs 0.5.0 with go-ipfs-dep 0.5.0 #1464

**Comment by lidel** [comment]
- recreated packages have the correct binary.
- we are now waiting for some admin around 𝕬𝖕𝖕𝖑𝖊 𝖓𝖔𝖙𝖆𝖗𝖞 𝖘𝖊𝖗𝖛𝖎𝖈𝖊

**Comment by NatoBoram** [comment]
Can the Linux & Windows binaries be published before the ones for MacOS?

**Comment by bertrandfalguiere** [comment]
> Thank you @bertrandfalguiere!
> 
> @jessicaschilling @rafaelramalho19
> I checked in with go-ipfs team and we are prioritizing shipping go-ipfs 0.5 to users, so we will cut 0.11.0 today with Desktop translation updates and go-ipfs 0.5 and later this week ship a patch release of Desktop (0.11.1) with fix for [#1287 (comment)](https://github.com/ipfs-shipyard/ipfs-desktop/issues/1287#issuecomment-620604299) and any missed translation updates in webui.

Apparently, traductions from [ipld-explorer-components](https://github.com/ipfs-shipyard/ipld-explorer-components) should be pulled and bubbled too, according to: https://github.com/ipfs-shipyard/ipfs-webui/issues/1477#issuecomment-620838432 
:)

**Comment by lidel** [comment]
@NatoBoram sadly the current setup (based on https://www.electron.build/auto-update.html) is all-or-nothing: we are unable to publish a release to the official channel without impacting Mac users.

That being said, if you want to help us testing the 0.11.0 on Linux, 
here are binaries that are waiting for publishing: 
`bafybeiegs4nplthqhneuhsm6n7qwczswf7yjr77yeimwbvkpi5mmewdrxm`

Let us know if anything looks odd, note the Tray issues (#1153)

**Comment by bertrandfalguiere** [comment]
@NatoBoram The Linux version was published a few hours ago on snap (now the default on Ubuntu 20.04).

@lidel is the realese on snap an independant/parallel process of/to realeases elsewhere?

**Comment by lidel** [comment]
IIRC it is a single build pipeline.

I think the update to the latest electron-builder changed something, and for some reason build does publish snap instead of attaching it to a github release draft. 

I was unable to find anything in release notes about his tho. Pretty scary change tbh.

**Comment by lidel** [comment]
Shipped: [v0.11.1](https://github.com/ipfs-shipyard/ipfs-desktop/releases/tag/v0.11.1) :ship:
